videoembedutility/twigextensions/VideoEmbedUtilityTwigExtension.php

```diff
@@ -108,13 +118,20 @@ public function videoEmbed($input, $options = array()) {
 						unset($options['class']);
 					}
 
-					$url .= '&' . http_build_query($options);
+                                        if(!empty($options)) {
+                                                $url .= '?' . http_build_query($options);
+                                        }
 				}
 
 				$originalPath = craft()->path->getTemplatesPath();
 				$myPath = craft()->path->getPluginsPath() . 'videoembedutility/templates/';
 				craft()->path->setTemplatesPath($myPath);
-				$markup = craft()->templates->render('_vimeoEmbed.html', array(
+                                if($this->videoHost($input) === WISTIA) {
```

Comment:
Instead of doing `if/else` here we should consider a service-to-template mapping. My php is *very* rusty so this is pseudocode, but something like:

```php
$serviceTemplates = [
  "WISTIA" => "_vistiaEmbed.html",
];

private function templateForService($service) {
  return $mapping[$service] ? $mapping[$service] : "_videoEmbed.html";
}

$templateHtml = this->templateForService($this->videoHost($input));
```